core/xml.js

```diff
@@ -186,8 +200,11 @@ Blockly.Xml.blockToDom = function(block, opt_noId) {
         container.appendChild(Blockly.Xml.cloneShadow_(shadow, opt_noId));
       }
       if (childBlock) {
-        container.appendChild(Blockly.Xml.blockToDom(childBlock, opt_noId));
-        empty = false;
+        var elem = Blockly.Xml.blockToDom(childBlock, opt_noId);
+        if (elem instanceof Element) {
```

Comment:
Instead compare the nodeType: 
`elem.nodeType == Blockly.utils.dom.NodeType.ELEMENT_NODE`

Reply:
Thank you! I really didn't want to use `instanceof` but I couldn't come up with a different check.